code/notificationbuilder/src/main/res/layout/push_tempate_vertical_catalog.xml

```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/catalog_container_layout"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:clipToPadding="false"
+    android:clipChildren="false"
+    android:orientation="vertical"
+    android:theme="@style/DayNightTheme">
+
+    <TextView
+        android:id="@+id/product_title"
+        style="@style/Title"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_gravity="start"
+        android:textAlignment="viewStart" />
+
+    <TextView
+        android:id="@+id/product_description"
+        style="@style/Body"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/product_title"
+        android:layout_gravity="start"
+        android:textAlignment="viewStart" />
+
+    <TextView
+        android:id="@+id/product_price"
+        style="@style/Body"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/product_description"
+        android:layout_marginTop="@dimen/product_catalog_price_top_margin"
+        android:layout_marginBottom="@dimen/product_catalog_price_bottom_margin"
+        android:layout_marginLeft="@dimen/product_catalog_price_side_margin"
+        android:layout_toRightOf="@id/product_image"
+        android:text="@string/price"
+        android:textAlignment="viewStart"
+        android:textStyle="normal|bold" />
+
+    <Button
+        android:id="@+id/cta_button"
+        android:layout_width="@dimen/product_catalog_cta_button_width"
+        android:layout_height="@dimen/product_catalog_cta_button_height"
+        android:layout_below="@id/product_price"
+        android:layout_marginLeft="@dimen/product_catalog_price_side_margin"
+        android:layout_toRightOf="@id/product_image"
+        android:text="@string/buy_now"
+        android:textAlignment="center"
+        android:textSize="12sp" />
+
+    <LinearLayout
+        android:id="@+id/vertical_catalog_thumbnail_layout"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_toRightOf="@id/product_image"
+        android:layout_below="@id/cta_button">
+
+        <ImageView
+            android:id="@+id/product_thumbnail_1"
+            android:scaleType="centerCrop"
+            android:layout_weight="1"
+            android:layout_height="@dimen/product_catalog_mini_image_height"
+            android:layout_width="@dimen/product_catalog_mini_image_width"
```

Comment:
You can set the layout_width to 0 for all thumbnail views instead of hardcoding the value. Since weight is 1 for all views in this layout, it will adjust based on available device width and make all of them be the same size